R/EstimationDataFromComparativeEffectivenessModel.R

```diff
@@ -0,0 +1,114 @@
+#' @param databaseId           A short string for identifying the database for query from comparative effectiveness
+#'                             data model (e.g. 'Synpuf')
+#' @param targetId             An integer identifying a cohort in exposure_id field of exposure_of_interest
+#'                             of OHDSI comparative effectiveness data model
+#' @param comparatorId         An integer identifying a cohort in exposure_id field of exposure_of_interest
+#'                             of OHDSI comparative effectiveness data model
+#' @param analysisId           An integer identifying the analysis_id field in cohort_method_analysis table
+#'                             of OHDSI comparative effectiveness data model
+#' @param formatName           (optional) Should the cohort name be formatted to remove prefixes and underscores? Default = FALSE
+#'
+#' @return                     tibble data frame
+#' 
+#' @examples                    SkeletonComparativeEffectStudy::getDataForPreferenceScorePlot(preferenceScoreDist = preferenceScoreDist, 
+#'                                                                                           exposureOfInterest = exposureOfInterest,
+#'                                                                                           targetId = 100,
+#'                                                                                           comparatorId = 200,
+#'                                                                                           databaseId = 'HCUP data',
+#'                                                                                           analysisId = 2)
+#' @export
+#'
+getDataForPreferenceScorePlot <- function(connection = NULL,
+                                          preferenceScoreDist = NULL,
+                                          exposureOfInterest = NULL,
+                                          targetId,
+                                          comparatorId,
+                                          outcomeId,
+                                          databaseId,
+                                          analysisId
+) {
+
+  
+  errorMessage <- checkmate::makeAssertCollection()
+  checkmate::assertInt(targetId, add = errorMessage)
+  checkmate::assertInt(comparatorId, add = errorMessage)
+  checkmate::assertInt(outcomeId, add = errorMessage)
+  checkmate::assertInt(analysisId, add = errorMessage)
+  checkmate::assertScalar(databaseId, add = errorMessage)
+  checkmate::assertCharacter(databaseId, add = errorMessage)
+
+  
+  if (!is.null(connection)) {
+    checkmate::reportAssertions(errorMessage)
+  } else if (is.null(connection)) {
+    checkmate::assertDataFrame(preferenceScoreDist, add = errorMessage)
+    checkmate::assertDataFrame(exposureOfInterest, add = errorMessage)
+    checkmate::reportAssertions(errorMessage)
+  } else {
+    stop("No connection or data frame.")
```

Comment:
This is a bit cryptic. Please refer to the arguments the user is supposed to provide, but didn't.